### PR TITLE
Setup Prow jobs for knative/caching

### PR DESCRIPTION
**Pull request by adrcunha**
Part of https://github.com/knative/caching/issues/1

### CODE REVIEW

File: ci/prow/config.yaml [hunk 1]
```diff
@@ -1789,6 +1953,68 @@ periodics:
...
+    containers:
+    - image: gcr.io/knative-tests/test-infra/coverage:latest
+      imagePullPolicy: Always
+      command:
+      - "/coverage"
+      args:
+      - "--artifacts=$(ARTIFACTS)"
+      - "--profile-name=coverage_profile.txt"
+      - "--cov-target=./pkg/"
+      - "--cov-threshold-percentage=80"
```
Comment (jessiezcc) [inline ci/prow/config.yaml]: 80 might be too high for a new repo?

Comment (adrcunha) [reply ci/prow/config.yaml]: It's not a blocker, so I don't think it's a problem.

Comment (jessiezcc) [reply ci/prow/config.yaml]: @steuhs, I see threshold parameter in post, continuous, pull coverage jobs. Which job will block PR, is it pull? If yes, 
what's the threshold usage in other post, continuous jobs?

Comment (adrcunha) [reply ci/prow/config.yaml]: And actually ALL repos have this threshold.

**Comment by adrcunha** [comment]
@jessiezcc Prow won't understand `/lgtm` if you add something after it :(

**Comment by jessiezcc** [comment]
opened https://github.com/kubernetes/test-infra/issues/9402 to track prow issu

**Pull request merged**